libqtile/widget/sensors.py

```diff
@@ -89,21 +88,24 @@ def get_temp_sensors(self):
                         kernel_module if kernel_module else "UNKNOWN", str(empty_index)
                     )
                     empty_index += 1
-                temperature_list[label] = (str(round(sensor.current, 1)), unit)
+                temperature_list[label] = sensor.current
 
         return temperature_list
 
     def poll(self):
         temp_values = self.get_temp_sensors()
         if temp_values is None:
             return False
-        text = ""
-        if self.show_tag and self.tag_sensor is not None:
-            text = self.tag_sensor + ": "
-        text += "".join(temp_values.get(self.tag_sensor, ["N/A"]))
-        temp_value = float(temp_values.get(self.tag_sensor, [0])[0])
+        temp_value = temp_values.get(self.tag_sensor, float("nan"))
         if temp_value > self.threshold:
             self.layout.colour = self.foreground_alert
         else:
             self.layout.colour = self.foreground_normal
-        return text
+        unit = "°C" if self.metric else "°F"
```

Comment:
We don't need to set this every time the widget is polled. Can we set once (e.g. during `_configure`) instead?